src/software/sensor_fusion/sensor_fusion.cpp

```diff
@@ -371,6 +378,63 @@ std::optional<Point> SensorFusion::getBallPlacementPoint(const SSLProto::Referee
     return point_opt;
 }
 
+void SensorFusion::updateDistanceDribbledByFriendlyTeam()
+{
+    // Dribble distance algorithm taken from TIGERs autoref implementation
+    // https://t.ly/vNZf9
+
+    if (!ball.has_value())
+    {
+        return;
+    }
+
+    // Add new touching robots
+    for (const Robot &robot : friendly_team.getAllRobots())
+    {
+        if (robot.isNearDribbler(ball->position()))
+        {
+            ball_contacts_by_friendly_robots.insert(
+                std::make_pair(robot.id(), ball->position()));
```

Comment:
a note that says that the insert only occurs if it doesn't exist would be helpful

Reply:
done